packages/react-native-performance/src/performance.ts

```diff
@@ -97,53 +137,59 @@ export const createPerformance = () => {
           `Failed to execute 'measure' on 'Performance': Cannot send start, end and duration options together.`
         );
       }
-    }
 
-    if (endMark) {
-      end = convertMarkToTimestamp(endMark);
-    } else if (startOrMeasureOptions && startOrMeasureOptions.end) {
-      end = convertMarkToTimestamp(startOrMeasureOptions.end);
-    } else if (
-      startOrMeasureOptions &&
-      startOrMeasureOptions.start &&
-      startOrMeasureOptions.duration
-    ) {
-      end =
-        convertMarkToTimestamp(startOrMeasureOptions.start) +
-        convertMarkToTimestamp(startOrMeasureOptions.duration);
-    } else {
-      end = now();
-    }
+      detail = startOrMeasureOptions.detail;
 
-    if (startOrMeasureOptions && startOrMeasureOptions.start) {
-      start = convertMarkToTimestamp(startOrMeasureOptions.start);
-    } else if (
-      startOrMeasureOptions &&
-      startOrMeasureOptions.end &&
-      startOrMeasureOptions.duration
-    ) {
-      start =
-        convertMarkToTimestamp(startOrMeasureOptions.end) -
-        convertMarkToTimestamp(startOrMeasureOptions.duration);
-    } else if (typeof startOrMeasureOptions === 'string') {
-      start = convertMarkToTimestamp(startOrMeasureOptions);
-    } else {
-      start = timeOrigin;
+      if (startOrMeasureOptions && startOrMeasureOptions.end) {
+        end = convertMarkToTimestamp(startOrMeasureOptions.end);
+      } else if (
+        startOrMeasureOptions &&
+        startOrMeasureOptions.start &&
+        startOrMeasureOptions.duration
+      ) {
+        end =
+          convertMarkToTimestamp(startOrMeasureOptions.start) +
+          convertMarkToTimestamp(startOrMeasureOptions.duration);
+      }
+
+      if (startOrMeasureOptions && startOrMeasureOptions.start) {
+        start = convertMarkToTimestamp(startOrMeasureOptions.start);
+      } else if (
+        startOrMeasureOptions &&
+        startOrMeasureOptions.end &&
+        startOrMeasureOptions.duration
+      ) {
+        start =
+          convertMarkToTimestamp(startOrMeasureOptions.end) -
+          convertMarkToTimestamp(startOrMeasureOptions.duration);
+      }
+    } else { 
+      if(endMark) {
+        end = convertMarkToTimestamp(endMark);
+      } else {
+        end = now();
+      }
+
+      if (typeof startOrMeasureOptions === 'string') {
+        start = convertMarkToTimestamp(startOrMeasureOptions);
+      } else {
+        start = timeOrigin;
+      }
     }
 
     return addEntry(
       new PerformanceMeasure(measureName, {
-        detail: startOrMeasureOptions
-          ? startOrMeasureOptions.detail
-          : undefined,
+        detail,
         startTime: start,
         duration: end - start,
       })
     );
   };
 
-  const metric = (name, valueOrOptions = {}) => {
```

Comment:
valueOrOptions can not be null, undefined or empty object. I guess here it is better to remove default.